base/sort.jl

```diff
@@ -1569,6 +1569,11 @@ The algorithms used by `DEFAULT_UNSTABLE` are an implementation detail. They are
 the same as those used by [`DEFAULT_STABLE`](@ref), but this is subject to change in future.
 """
 const DEFAULT_UNSTABLE = DEFAULT_STABLE
+"""
+    SMALL_THRESHOLD
+    
+SMALL_THRESHOLD is a constant that is used internally to decide what is considered to be small or large.    
```

Comment:
That's a lot better already, but it isn't yet all that helpful as it gives no indication about the context in which it is used, i.e. what "small" or "large" mean or affect.

I think something to the effect of: for vectors with less (?) than `SMALL_THRESHOLD` elements, sorting switches to a different algorithm `SMALL_ALGORITHM`, otherwise it uses the `big` algorithm. This is in particular relevant for recursive sorting algorithms which reduce to smaller sublists, sort those and combine the result.


Note that `SMALL_ALGORITHM` also has a docstring.

Reply:
The threshold at which to switch sorting algorithms based on size is also quite variable. SMALL_THRESHOLD is only used in some cases (e.g. QuickSort and MergeSort)